snap/info.go

```diff
@@ -643,28 +644,69 @@ func (s *Info) CommonDataSaveDir() string {
 	return CommonDataSaveDir(s.InstanceName())
 }
 
+// Get home directories from system-params file.
+// Home directories are returned as a ',' seperated string
+func HomeDirsSysParams(rootDir string) (string, error) {
+
+	ssp, err := sysparams.Open(rootDir)
```

Comment:
it's not reasonable to re-read this file all the time from these helpers, we need to find a way to early and on changes propagate this value into this package and other relevant places

Reply:
one option could be to add a way in the dirs package to cache our notions of what are the homedirs, and then set up the values from somewhere like configstate.Init and also update it from configcore